src/handler/sessions/session.rs

```diff
@@ -66,6 +83,8 @@ impl Session {
 
         // If the message nonce length is ever set below 4 bytes this will explode. The packet
         // size constants shouldn't be modified.
+        const _: () = assert!(MESSAGE_NONCE_LENGTH > 4);
```

Comment:
this will be optimized by the compiler to an empty constant so it won't be called everytime. But agreed, this should be moved outside the function, as a normal constant